spec/models/spree/product_spec.rb

```diff
@@ -40,7 +40,8 @@ module Spree
 
         it "generates a unique permalink" do
           product1 = create(:product, name: "Banana Permanenta", permalink: nil)
-          product2 = create(:product, name: "Banana Permanenta", permalink: nil)
+          product2 = build_stubbed(:product, name: "Banana Permanenta", permalink: nil)
+          expect(product2).to be_valid
```

Comment:
This is required to trigger the [`before_validation`](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/models/spree/product_decorator.rb#L31) callback which sets the permalink.

Reply:
have you tried with [run_validations](https://apidock.com/rails/v4.0.2/ActiveModel/Validations/run_validations%21)? I bet it'll trigger the callback and it'd be slightly more explicit than an expectation, which is not what we want.

Reply:
It's a `protected` method 😞